### PR TITLE
retry() dissabled

### PR DESCRIPTION
**Pull request by nedievas**
tested 'zenbot trade --debug bitfinex.btc-usd' without issues.

### CODE REVIEW

**Review by crubb — COMMENTED** [review]
At least on Kraken it's extremely sensible to use the `retry()` function and nobody can tell me bitfinex is 100% stable ;)

I vote for the `retry()` function to stay as it is.

If bitfinex is stable it won't be called.

File: extensions/exchanges/bitfinex/exchange.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ module.exports = function container (get, set, clear) {
       exchange[method].apply(exchange, args)
     }, 10000)
   }
-
+*/
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:38]: no harm in leaving it in then

File: extensions/exchanges/bitfinex/exchange.js [hunk 2]
```diff
@@ -48,11 +48,12 @@ module.exports = function container (get, set, clear) {
     },
 
     getTrades: function (opts, cb) {
-      var func_args = [].slice.call(arguments)
+//      var func_args = [].slice.call(arguments)
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:51]: this is getting quite ugly, if it's not necessary then please just remove it, we have git history

File: extensions/exchanges/bitfinex/exchange.js [hunk 3]
```diff
@@ -48,11 +48,12 @@ module.exports = function container (get, set, clear) {
...
       var client = publicClient()
       var args = joinProduct(opts.product_id)
       client.trades(args, function (err, body) {
-        if (err) return retry('getTrades', func_args, err)
+//        if (err) return retry('getTrades', func_args, err)
+if (err) return(err)
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:56]: indenting and echoing the same comment

File: extensions/exchanges/bitfinex/exchange.js [hunk 4]
```diff
@@ -67,10 +68,11 @@ module.exports = function container (get, set, clear) {
     },
 
     getBalance: function (opts, cb) {
-      var func_args = [].slice.call(arguments)
+//      var func_args = [].slice.call(arguments)
       var client = authedClient()
       client.wallet_balances(function (err, body) {
-        if (err) return retry('getBalance', func_args, err)
+//        if (err) return retry('getBalance', func_args, err)
+        if (err) return(err)
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:75]: echoing previous comments

File: extensions/exchanges/bitfinex/exchange.js [hunk 5]
```diff
@@ -87,26 +89,28 @@ module.exports = function container (get, set, clear) {
...
 
     getQuote: function (opts, cb) {
-      var func_args = [].slice.call(arguments)
+//      var func_args = [].slice.call(arguments)
       var client = publicClient()
       var pair = joinProduct(opts.product_id)
       client.ticker(pair, function (err, body) {
-        if (err) return retry('getQuote', func_args, err)
+//        if (err) return retry('getQuote', func_args, err)
+if (err) return(err)
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:97]: echoing previous comment

File: extensions/exchanges/bitfinex/exchange.js [hunk 6]
```diff
@@ -87,26 +89,28 @@ module.exports = function container (get, set, clear) {
...
     },
 
     cancelOrder: function (opts, cb) {
-      var func_args = [].slice.call(arguments)
+//      var func_args = [].slice.call(arguments)
       var client = authedClient()
       client.cancel_order(opts.order_id, function (err, body) {
-        if (err) return retry('cancelOrder', func_args, err)
+//        if (err) return retry('cancelOrder', func_args, err)
+if (err) return(err)
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:107]: echoing previous comment

File: extensions/exchanges/bitfinex/exchange.js [hunk 7]
```diff
@@ -136,9 +140,10 @@ module.exports = function container (get, set, clear) {
         is_postonly
       }
       client.make_request('order/new', params, function (err, body) {
-        if (body.is_live === true) {
+                console.log(body)
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:143]: console.log?

File: extensions/exchanges/bitfinex/exchange.js [hunk 8]
```diff
@@ -136,9 +140,10 @@ module.exports = function container (get, set, clear) {
...
+//        if (body && body.is_live === true) {
           var order = {
-            id: body.id,
+            id: body && body.is_live === true ? body.order_id : null,
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:146]: indenting, this seems to be the only good fix

File: extensions/exchanges/bitfinex/exchange.js [hunk 9]
```diff
@@ -147,20 +152,21 @@ module.exports = function container (get, set, clear) {
...
-        }
+//        }
         if (err && err.toString('Error: Invalid order: not enough exchange balance')) {
           status: 'rejected'
           reject_reason: 'balance'
           return cb(null, order)
         }
-        if (err) return retry('buy', func_args, err)
+  //      if (err) return retry('buy', func_args, err)
+  if (err) return(err)
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:162]: echoing previous comment

File: extensions/exchanges/bitfinex/exchange.js [hunk 10]
```diff
@@ -190,35 +196,37 @@ module.exports = function container (get, set, clear) {
...
+//        if (body && body.is_live === true) {
+        var order = {
+          id: body && body.is_live === true ? body.order_id : null,
+          status: 'open',
+          price: opts.price,
+          size: opts.size,
+          post_only: !!opts.post_only,
+          created_at: new Date().getTime(),
+          filled_size: '0',
+          ordertype: c.order_type
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:208]: indenting

File: extensions/exchanges/bitfinex/exchange.js [hunk 11]
```diff
@@ -228,6 +236,12 @@ module.exports = function container (get, set, clear) {
           order.done_at = new Date().getTime()
           return cb(null, order)
         }
+        if (body.is_cancelled  === true && err.toString('Postonly Canceled')) {
+          order.status = 'rejected'
+          order.reject_reason = 'post only'
+          order.done_at = new Date().getTime()
+          return cb(null, order)
+        }
```
Comment (DeviaVir) [inline extensions/exchanges/bitfinex/exchange.js:244]: this is also one of the better parts of this PR

**Review by DeviaVir — CHANGES_REQUESTED** [review]
this is quite messy for no real reason.. please use eslint

**Pull request closed without merging**